File: Data Structures & Algorithms/anagram-groups/submission-3.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<Integer, List<String>> resultMap = new HashMap<>();

        for (String s : strs) {
            int[] count = new int[26];
            for (char c : s.toCharArray()) {
                count[c - 'a']++;
            }
            int arrayHash = Arrays.hashCode(count);
            resultMap.putIfAbsent(arrayHash, new ArrayList<>());
            resultMap.get(arrayHash).add(s);
        }

        return new ArrayList<>(resultMap.values());
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-3.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res = ""
        for s in strs:
            res += str(len(s)) + "#" + s
        return res

    def decode(self, s: str) -> List[str]:
        res = []
        i = 0

        while i < len(s):
            j = i
            while s[j] != '#':
                j += 1
            
            print("j: ", j)
            length = int(s[i:j])
            print("length: ", length)
            j += 1
            i = j
            j += length
            st = s[i:j]
            res.append(st)
            i = j

        return res